autoUniCalendar.py

```diff
@@ -2,156 +2,430 @@
+        "subjects": {},
+        "Q1": [0, 0],
+        "Q2": [0, 0]
+    }
+
     # Separate the events from its XML context.
-    text = f.read().split('<')
+    text = rawResponse.split('<')
     events = text[5].split('{')
     del events[0:2]
 
+    if icsMode:
+        c = Calendar()
+    elif not dryRun:
+        g = open(filename + ".csv", "w")
+        g.write("Subject,Start Date,Start Time,End Date,End Time,Location,Description\n")
+
     # Each field of the event is separated by commas.
-    print("[*] Parsing the data...")
     for event in events:
         data = []
         for field in event.split(','):
             # Remove empty fields.
             if field.strip():
                 data.append(field)
         # Save in variables the fields needed to build the CSV line of the event.
-        title = data[1]
-        start = data[2]
-        end = data[3]
-        description = data[7]
-        
-        # Make the necessary strings transformations to adapts the raw field data into a CSV readable file.
-        title_csv = re.findall('"([^"]*)"', title.split(':')[1])[0]
-        start_date = start.split(' ')[1].split('T')[0].split('"')[1]
-        start_date_csv = start_date.split('-')[2]+'/'+start_date.split('-')[1]+'/'+start_date.split('-')[0]
-        start_hour = start.split(' ')[1].split('T')[1].split('+')[0]
-        end_date = end.split(' ')[1].split('T')[0].split('"')[1]
-        end_date_csv = end_date.split('-')[2]+'/'+end_date.split('-')[1]+'/'+end_date.split('-')[0]
-        end_hour = end.split(' ')[1].split('T')[1].split('+')[0]
-        alert_date = start_date_csv
-        alert_hour = str(int(start.split(' ')[1].split('T')[1].split('+')[0].split(':')[0]) - 1) + ':' + start.split(' ')[1].split('T')[1].split('+')[0].split(':')[1] + ':' + start.split(' ')[1].split('T')[1].split('+')[0].split(':')[2]
-        event_creator = "Universidad de Oviedo"
-        body = description.split('"')[3].replace(r'\n', '')
-        # Write all the fields into a single line, and append it to the file.
-        csv_line = f"{title_csv},{start_date_csv},{start_hour},{end_date_csv},{end_hour},FALSO,FALSO,{alert_date},{alert_hour},{event_creator},,,,,,{body},,,Normal,Falso,Normal,2\n"
-        g.write(csv_line)
-        
-    print("[*] Events correctly written in the CSV file.")
-    f.close()
-    g.close()
-    print("[*] Removing raw .txt file...")
-    os.remove("raw.txt")
-    print("\n[#] Calendar generated. You can now import it in Outlook or Google Calendar selecting 'import from file' and providing the CSV file generated.\n")
-
-
-first_request = get_first_request(session, render_map)
-cookies = extract_cookies(first_request)
-post_second_request(session, render_map, "true", cookies[0], cookies[1], "1662444000000", "1683612000000", cookies[2])
-create_csv("raw.txt")
\ No newline at end of file
+        uid = data[0].split('": "')[1].replace('"', '')
+        title = data[1].split('": "')[1].replace('"', '')
```

Comment:
Can break if the title of the course has ": " in the name.

Reply:
very similar to your own commit earlier today (https://github.com/Bimo99B9/autoUniCalendar/commit/f4f2c621ce15d5dd4fba29f3a56f4ef610a1f3b8). can only break if the title includes `": "` in succession, which is highly improbable.

Reply:
solved in https://github.com/Bimo99B9/autoUniCalendar/pull/6/commits/d54089c101a069c6737d233a4c318d8f8ca879fa, pretty sure this is the safest way to parse titles and such.